flang/include/flang/Lower/AbstractConverter.h

```diff
@@ -91,6 +91,19 @@ class AbstractConverter {
     return genExprValue(*someExpr, &loc);
   }
 
+  /// Generate the address of the box describing the variable designated
+  /// by the expression. The expression must be an allocatable or pointer
+  /// designator.
+  virtual fir::BoxAddressValue
```

Comment:
`BoxInstanceValue ` is not clearer to me, but I see your point. I actually used `PointerOrAllocatable` in the SymBox renaming of the fir BoxValues.
We could maybe call it a `MutableBoxValue` since the main property of this category is that the box is in memory and its content may be modified during the program.
What do you think about it ?

Reply:
Some drive by possibilities;
  AllocatableOrPointerBox
  AllocOrPtrBox

Reply:
> `BoxInstanceValue ` is not clearer to me, but I see your point. I actually used `PointerOrAllocatable` in the SymBox renaming of the fir BoxValues.

+1

> We could maybe call it a `MutableBoxValue` since the main property of this category is that the box is in memory and its content may be modified during the program.
> What do you think about it ?

Exactly: its a box value in memory, thus mutable, addressable, etc. ("Instance" implies memory resident to me, but that may be a reflection of my background. Shrug.) Memory or mutable are improvements over address to me. Although "cashbox" would be a nice pun.